pydoctor/test/__init__.py

```diff
@@ -1,6 +1,22 @@
 """PyDoctor's test suite."""
 
+from logging import LogRecord
+from typing import NamedTuple, Sequence
 import sys
 import pytest
 
 typecomment = pytest.mark.skipif(sys.version_info < (3, 8), reason="requires python 3.8")
+
+
+# Because pytest 6.1 does not yet export types for fixtures, we define
+# approximations that are good enough for our test cases:
+
+class CapLog:
+    records: Sequence[LogRecord]
+
+class CaptureResult(NamedTuple):
+    out: str
+    err: str
+
+class CapSys:
+    def readouterr(self) -> CaptureResult: ...
```

Comment:
might be better to use Protocols here, rather than concrete types - as that's what the pytest fixture types in `pytest.` will be.

otherwise you can use the exported types in `_pytest.logging.LogCaptureFixture`

```suggestion
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing_extensions import Protocol

    class CapLog(Protocol):
        records: Sequence[LogRecord]

    class CaptureResult(Protocol):
        out: str
        err: str

    class CapSys(Protocol):
        def readouterr(self) -> CaptureResult: ...
else:
    CapLog = CaptureResult = CapSys = None
```

Reply:
That's probably a good idea, for #219 though.